app/gui2/src/components/GraphEditor/widgets/WidgetVector.vue

```diff
@@ -6,21 +6,43 @@ import { ForcePort } from '@/providers/portInfo'
 import type { WidgetInput } from '@/providers/widgetRegistry'
 import { Score, defineWidget, widgetProps } from '@/providers/widgetRegistry'
 import { useGraphStore } from '@/stores/graph'
+import type { SuggestionEntryArgument } from '@/stores/suggestionDatabase/entry'
 import { Ast, RawAst } from '@/util/ast'
+import { ArgumentAst } from '@/util/callTree'
 import { computed } from 'vue'
 
 const props = defineProps(widgetProps(widgetDefinition))
+const inputAst = computed(() =>
+  props.input instanceof ArgumentAst ? props.input.ast : props.input,
+)
+const argInfo = computed(() => (props.input instanceof ArgumentAst ? props.input.info : undefined))
+
+const defaultConstructor = computed(() => {
+  const fallback = Ast.Wildcard.new
+  const config = props.config
+  if (config == null) return fallback
+  const [_, widgetConfig] = config.find(([name]) => name === argInfo.value?.name) ?? []
+  if (
+    widgetConfig &&
+    widgetConfig.kind == 'Vector_Editor' &&
+    widgetConfig.item_editor.kind === 'Single_Choice'
```

Comment:
Why the item_editor must be a `Single_Choice`? I thought the default can be just any value, e.g. number for number slider.